core/types/transaction.go

```diff
@@ -784,6 +788,23 @@ type MevBundle struct {
 	Hash              common.Hash
 }
 
+func (b *MevBundle) UniquePayload() []byte {
+	var buf []byte
+	buf = binary.AppendVarint(buf, b.BlockNumber.Int64())
+	buf = append(buf, b.Hash[:]...)
+	sort.Slice(b.RevertingTxHashes, func(i, j int) bool {
+		return bytes.Compare(b.RevertingTxHashes[i][:], b.RevertingTxHashes[j][:]) <= 0
+	})
+	for _, txHash := range b.RevertingTxHashes {
+		buf = append(buf, txHash[:]...)
+	}
+	return buf
+}
+
+func (b *MevBundle) ComputeUUID() uuid.UUID {
```

Comment:
suggestion (non-blocking)

We should probably event use a `sync.Once` pattern to initialize the UUID once since we don't mutate bundles after initialization

Reply:
Yeah, I thought in a little bit different direction, I believe we'll eventually move to propagating this uuid as a field of bundle structure, but it'd be better be done once we have some module with shared models. But I anyway understand your concern, I hope we'll address it in future